### PR TITLE
Sgratzl/datalimits

### LINKED ISSUE #13: Reduce range of displayed outliers / do not display outliers
Hi,

first of all thanks a lot for this plug-in, it works great!
I have one feature request though: would it be possible to not display outliers / reduce the range of displayed outliers? Otherwise, if there are outliers far away, it can make the plot a lot less readable (see attached). What I am most interested in are the median calculated quartiles; I am happy to chop off the fourth quartile at a certain point, knowing that there are outliers above. So if I could somehow set a flag to suppress outliers, that would be perfect.

Thanks!

screenshot from 2018-07-13 08-50-06

### PR DESCRIPTION
**Pull request by sgratzl**
fix #13 option to different min/max data stats

### CODE REVIEW

**Comment by sgratzl** [comment]
@sluger ping

File: README.md [hunk 1]
```diff
@@ -95,7 +95,29 @@ interface IViolinStyling extends IBaseStyling {
...
+		 * statistic measure that should be used for computing the minimal data limit
+		 * @default 'min'
+		 */
+		minStats: 'min'|'q1'|'whiskerMin';
+
+		/**
+		 * statistic measure that should be used for computing the maximal data limit
+		 * @default 'max'
+		 */
+		minStats: 'max'|'q3'|'whiskerMax';
```
Comment (sluger) [inline README.md]: copy paste error i guess, this should be called `maxStats`

File: README.md [hunk 2]
```diff
@@ -108,34 +130,34 @@ interface IBaseItem {
...
   q1: number;
   q3: number;
   /**
-   * list of box plot outlier values  
+   * list of box plot outlier values
    */
   outliers?: number[];
 }
 
 interface IKDESamplePoint {
```
Comment (sluger) [inline README.md]: I had to look up what KDE means, maybe a short note in the readme that the Kernel density estimation is used here would help

File: samples/utils.js [hunk 3]
```diff
@@ -47,12 +47,16 @@ window.chartColors = {
...
 		},
 
-		rand: function(min, max) {
+		randF: function(min, max) {
 			var seed = this._seed;
 			min = min === undefined ? 0 : min;
 			max = max === undefined ? 1 : max;
 			this._seed = (seed * 9301 + 49297) % 233280;
-			return min + (this._seed / 233280) * (max - min);
+			return () => min + (this._seed / 233280) * (max - min);
```
Comment (sluger) [inline samples/utils.js]: can you elaborate on the change to return a function instead?

File: samples/utils.js [hunk 4]
```diff
@@ -66,10 +70,12 @@ window.chartColors = {
 			var dfactor = Math.pow(10, decimals) || 0;
 			var data = [];
 			var i, value;
+			var rand = cfg.random ? cfg.random(min, max) : this.randF(min, max);
+			var rand01 = cfg.random01 ? cfg.random01() : this.randF();
```
Comment (sluger) [inline samples/utils.js]: random01 does not appear anywhere else, where does it come from, is it a typo or unused?

File: samples/datalimits.html [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+				var random = (min, max) => () => (b() * ((max || 1) - (min || 0))) + (min || 0);
+				// var random = d3.randomUniform;
+				var boxplotData = {
+            labels: samples.months({count: 7}),
+            datasets: [{
+                label: 'Dataset 1',
+                backgroundColor: color(window.chartColors.red).alpha(0.5).rgbString(),
+                borderColor: window.chartColors.red,
+                borderWidth: 1,
+                data: samples.boxplots({count: 7, random: random}),
```
Comment (sluger) [inline samples/datalimits.html]: use shorthand assignment: `{count: 7, random}`

File: samples/datalimits.html [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+                outlierColor: '#999999',
+                padding: 10
+            }, {
+                label: 'Dataset 2',
+                backgroundColor: color(window.chartColors.blue).alpha(0.5).rgbString(),
+                borderColor: window.chartColors.blue,
+                borderWidth: 1,
+                data: samples.boxplotsArray({count: 7, random: random}),
```
Comment (sluger) [inline samples/datalimits.html]: use shorthand assignment: `{count: 7, random}`

File: samples/utils.js (hunk 3 above)
Comment (sgratzl) [reply samples/utils.js]: will be changed in https://github.com/datavisyn/chartjs-chart-box-and-violin-plot/pull/16/files again

File: samples/utils.js (hunk 4 above)
Comment (sgratzl) [reply samples/utils.js]: there are two random number generators, one for generating the data and one for generating missing values by chance = 0..1 in some cases one just wanna change one of them but not the other. see violin example

File: samples/datalimits.html (hunk 5 above)
Comment (sgratzl) [reply samples/datalimits.html]: do we wanna use full ES6 syntax here?

Comment (sluger) [reply samples/datalimits.html]: we do not necessarily have to

**Pull request merged**